test/maven/transform/addPluginTransform.test.ts

```diff
@@ -0,0 +1,136 @@
+/*
+ * Copyright © 2018 Atomist, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {
+    InMemoryProject,
+    InMemoryProjectFile,
+    Project,
+} from "@atomist/automation-client";
+import * as xml from "fast-xml-parser";
+import * as assert from "power-assert";
+import { addPluginTransform } from "../../../lib/maven/transform/addPluginTransform";
+
+describe("addPluginTransformTest", () => {
+
+    it("no POM causes no change, no error", async () => {
+        const dae = {
+            group: "bigco",
+            artifact: "software",
+            name: "software",
+            version: "2.0.0",
+        };
+        // const pom = PomWithReplaceMe.replace("replaceMe", "");
+        const p = InMemoryProject.of();
+        await addPluginTransform(dae)(p, null, null);
+        assert(!(await p.getFile("pom.xml")));
+    });
+
+    it("makes addition when plugin not present", async () => {
+        const dae = {
+            group: "bigco",
+            artifact: "software",
+            name: "software",
+            version: "2.0.0",
+            configuration: {
+                foo: {
+                    bar1: "hello",
+                },
+                baz: "world",
+            },
+        };
+        const p = projectOfPom(Pom);
+        await addPluginTransform(dae)(p, null, null);
+        const newPomContent = p.findFileSync("pom.xml").getContentSync();
+        // console.log(newPomContent);
+        assert(xml.validate(newPomContent));
+        assert(newPomContent.indexOf(dae.group) !== -1);
+        assert(newPomContent.indexOf(dae.artifact) !== -1);
+        assert(newPomContent.indexOf(dae.version) !== -1);
+        assert(newPomContent.indexOf("<foo>") !== -1);
```

Comment:
You decided not to test the formatting?
this is legitimate; the formatting being different is not the end of the world and it could be a fragile test